PhysicsTools/Scouting/plugins/Run3ScoutingElectronBestTrack.cc

```diff
@@ -0,0 +1,216 @@
+  produces<edm::ValueMap<float>>("Run3ScoutingElectronTrackpt");
+  produces<edm::ValueMap<float>>("Run3ScoutingElectronTracketa");
+  produces<edm::ValueMap<float>>("Run3ScoutingElectronTrackphi");
+  produces<edm::ValueMap<float>>("Run3ScoutingElectronTrackpMode");
+  produces<edm::ValueMap<float>>("Run3ScoutingElectronTracketaMode");
+  produces<edm::ValueMap<float>>("Run3ScoutingElectronTrackphiMode");
+  produces<edm::ValueMap<float>>("Run3ScoutingElectronTrackqoverpModeError");
+  produces<edm::ValueMap<float>>("Run3ScoutingElectronTrackchi2overndf");
+  produces<edm::ValueMap<int>>("Run3ScoutingElectronTrackcharge");
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void Run3ScoutingElectronBestTrackProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::Handle<std::vector<Run3ScoutingElectron>> run3ScoutingElectronHandle;
+  iEvent.getByToken(run3ScoutingElectronToken_, run3ScoutingElectronHandle);
+
+  if (!run3ScoutingElectronHandle.isValid()) {
+    // Handle the absence as a warning
+    edm::LogWarning("Run3ScoutingElectronBestTrackProducer")
+        << "No Run3ScoutingElectron collection found in the event!";
+    return;
+  }
+
+  const size_t num_electrons = run3ScoutingElectronHandle->size();
+  std::vector<int> besttrk_idx(num_electrons, -1);
+  std::vector<float> besttrk_d0s(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_dzs(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_pts(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_etas(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_phis(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_pModes(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_etaModes(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_phiModes(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_qoverpModeErrors(num_electrons, std::numeric_limits<float>::max());
+  std::vector<float> besttrk_chi2overndfs(num_electrons, std::numeric_limits<float>::max());
+  std::vector<int> besttrk_charges(num_electrons, std::numeric_limits<int>::max());
+
+  for (size_t iElectron = 0; iElectron < num_electrons; ++iElectron) {
+    const Run3ScoutingElectron& electron = run3ScoutingElectronHandle->at(iElectron);
+    const math::PtEtaPhiMLorentzVector cluster(electron.pt(), electron.eta(), electron.phi(), 0.0005);
+
+    double besttrack_ediff = std::numeric_limits<double>::max();
+
+    for (unsigned int i = 0; i < electron.trkpt().size(); ++i) {
+      const unsigned int eta_idx = (abs(electron.trketa().at(i)) < 1.479) ? 0 : 1;
+      if (electron.trkpt().at(i) < trackPtMin_.at(eta_idx))
+        continue;
+      if (electron.trkchi2overndf().at(i) > trackChi2OverNdofMax_.at(eta_idx))
```

Comment:
just wondering, here and in many places, do you really need (slow) bounds-check access with `.at()` or would the subscript operator `[]` enough for the purpose?

Reply:
I will change to subscript operator. I was not aware that one is slower than the other.